### DISCUSSION TITLE
Feature request: spectral normalization

### DISCUSSION THREAD

**Discussion by rdilip** [google/flax#2902] [Ideas] [answered]
Are spectral normalization layers something that anyone's working on at the moment? They're present in Haiku, so there's a template to work off of, and they'd be very useful (e.g., to enable things like https://www.tensorflow.org/tutorials/understanding/sngp).

**Comment by cgarciae** [comment]
@chiamp recently added #2881, maybe this could also be added?

**Comment by dnajera27** [comment]
I am also interested in leveraging spectral normalization. I was wondering if there had been any work done to implement SNGP since this was posted. Thanks!

**Comment by chiamp** [comment, marked as answer]
Spectral norm is now available after #3335. Check out the [API docs](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/layers.html#flax.linen.SpectralNorm).
